File: app/src/main/java/com/mdgd/pokemon/models/repo/PokemonsRepo.java
package com.mdgd.pokemon.models.repo;

import com.mdgd.pokemon.models.infra.Result;
import com.mdgd.pokemon.models.repo.dao.schemas.PokemonFullDataSchema;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.Single;

public interface PokemonsRepo {
    int PAGE_SIZE = 30;
    long LOADING_COMPLETE = -1;

    Single<Result<List<PokemonFullDataSchema>>> getPage(Integer page);

    Observable<Result<Long>> loadPokemons();
}

File: models/src/main/java/com/mdgd/pokemon/models/repo/PokemonsRepo.kt
package com.mdgd.pokemon.models.repo

import com.mdgd.pokemon.models.repo.dao.schemas.PokemonFullDataSchema

interface PokemonsRepo {
    fun getPokemons(): List<PokemonFullDataSchema>

    suspend fun getPage(page: Int): List<PokemonFullDataSchema>
    suspend fun loadPokemons(initialAmount: Long): Long
    suspend fun getPokemonById(pokemonId: Long): PokemonFullDataSchema?
    suspend fun loadInitialPages(amount: Long)

    companion object {
        const val PAGE_SIZE = 30
    }
}
